src/ServiceManagerBase.sol

```diff
@@ -135,6 +136,145 @@ abstract contract ServiceManagerBase is ServiceManagerBaseStorage {
         _setRewardsInitiator(newRewardsInitiator);
     }
 
+    function migrateToOperatorSets() external onlyOwner {
```

Comment:
Yeah i think for EigenDA it could be an issue atm.  I ran some gas tests for 4 quorums and ~100 operators per quorum and it was on the order of 100s million gas.  I think the function could be optimized though since i haven't really focused on that yet.

I refactored to a view function like you suggested but left it being called internally to the `migrate()` for now to see if we can optimize to get to something reasonable.  If we can't get it down to around 10 million gas then we should do multiple txs

Reply:
I did a bit more testing this morning to split out the view and pass in the state that needed to be updated vs. doing the view calls in the state-changing function, and it's ~21 million gas based on 3 quorums and 200 operators per quorum. I consider this worst case scenario if EigenDA has reAlt go live on mainnet and they have about the number of operators in the Eigen and Eth quorums

Reply:
Migrating can now be broken up over multiple transactions if the situation arises to prevent potential issues with gas costs exceed the block size
[3fcc68e](https://github.com/Layr-Labs/eigenlayer-middleware/pull/286/commits/3fcc68e1370cf86c893790f0e44d62e7ad714e7d)